src/compiler/scala/tools/nsc/backend/jvm/ClassfileWriters.scala

```diff
@@ -236,10 +236,18 @@ abstract class ClassfileWriters {
     val noAttributes = Array.empty[FileAttribute[_]]
     private val isWindows = scala.util.Properties.isWin
 
+    private def checkName(component: Path): Unit = if (isWindows) {
+      val specials = raw"(?i)CON|PRN|AUX|NUL|COM[1-9]|LPT[1-9]".r
+      val name = component.toString
+      def warnSpecial(): Unit = frontendAccess.backendReporting.warning(NoPosition, s"path component is special Windows device: ${name}")
+      specials.findPrefixOf(name).foreach(prefix => if (prefix.length == name.length || name(prefix.length) == '.') warnSpecial())
+    }
+
```

Comment:
it is windows but could still use a test.  I think the request was always to warn, but almost everything is distributed via jar or zip. Unzip at your own risk.